### PR TITLE
docs: add docs for serverInfo and connector functions in java client

### PR DESCRIPTION
**Pull request by jzaralim**
### Description 
Docs for the following java client functions: serverInfo, createConnector, dropConnector, listConnectors and describeConnector.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:696]: We should update the table of contents at the top of this file with the new sections so they're more discoverable.

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:697]:
```suggestion
-------------------------------------
```

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:723]: Add a link to the API reference, either here or for the individual methods.

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:736]:
```suggestion
client.createConnector("jdbc-connector", true, connectorProperties).get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:731]: nit: inconsistent spacing in this example -- add a space on this line and the ones below?

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:741]:
```suggestion
client.dropConnector("jdbc-connector").get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:746]:
```suggestion
List<ConnectorInfo> connectors = client.listConnectors().get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:751]: What are possible values returned by this? We don't need to add it here but we should update the javadoc for this method to include this information.

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:758]:
```suggestion
ConnectorDescription description = client.describeConnector("jdbc-connector").get();
```

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:760]: nit: add newline after the periods in this output.

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:715]:
```suggestion
------------------------------------
```

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (vcrfxia) [inline docs/developer-guide/ksqldb-clients/java-client.md:699]: Ideally this piece would be in a separate PR so we can get the change live for 0.16 ahead of the connector docs, but it's not a huge deal; your call on whether you think it's worth the effort.

**Review by vcrfxia — COMMENTED** [review]
Thanks @jzaralim -- bunch of comments inline but LGTM overall!

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (JimGalasyn) [reply docs/developer-guide/ksqldb-clients/java-client.md:699]: Joel and I had a chat with MD and Apurva, and we're trying to get me out of the Mechanical Turk business. :)

**Review by JimGalasyn — APPROVED** [review]
LGTM, modulo suggestions by @vcrfxia.

**Review by vcrfxia — APPROVED** [review]
🎉 Thanks @jzaralim !

File: docs/developer-guide/ksqldb-clients/java-client.md
Comment (jzaralim) [reply docs/developer-guide/ksqldb-clients/java-client.md]: Since the rest of the docs will be coming out for 0.18, I'll move this to another PR.

**Pull request merged**
